src/main/scala/com/gu/scanamo/ScanamoRequest.scala

```diff
@@ -41,22 +41,22 @@ object ScanamoRequest {
     * prop> import com.amazonaws.services.dynamodbv2.model._
     *
     * prop> (keyName: String, keyValue: Long, tableName: String) =>
-    *     |   val deleteRequest = ScanamoRequest.deleteRequest(tableName)(keyName -> keyValue)
+    *     |   val deleteRequest = ScanamoRequest.deleteRequest(tableName)(Symbol(keyName) -> keyValue)
     *     |   deleteRequest.getTableName == tableName &&
     *     |   deleteRequest.getKey == Map(keyName -> new AttributeValue().withN(keyValue.toString)).asJava
     * }}}
     */
-  def deleteRequest[K](tableName: String)(key: (String, K)*)(implicit fk: DynamoFormat[K]): DeleteItemRequest =
+  def deleteRequest[K](tableName: String)(key: (Symbol, K)*)(implicit fk: DynamoFormat[K]): DeleteItemRequest =
     new DeleteItemRequest().withTableName(tableName).withKey(asAVMap(key: _*))
 
-  def queryRequest[K](tableName: String)(key: (String, K))(implicit fk: DynamoFormat[K]): QueryRequest = {
+  def queryRequest[K](tableName: String)(key: (Symbol, K))(implicit fk: DynamoFormat[K]): QueryRequest = {
     val (k, v) = key
     new QueryRequest().withTableName(tableName)
-      .withKeyConditionExpression(s"#K = :$k")
-      .withExpressionAttributeNames(Map("#K" -> k).asJava)
-      .withExpressionAttributeValues(asAVMap(s":$k" -> v))
+      .withKeyConditionExpression(s"#K = :${k.name}")
+      .withExpressionAttributeNames(Map("#K" -> k.name).asJava)
+      .withExpressionAttributeValues(asAVMap(Symbol(s":${k.name}") -> v))
   }
 
-  private def asAVMap[K](kvs: (String, K)*)(implicit fk: DynamoFormat[K]) =
-    Map(kvs: _*).mapValues(fk.write).asJava
+  private def asAVMap[K](kvs: (Symbol, K)*)(implicit fk: DynamoFormat[K]) =
+    Map(kvs: _*).map { case (k,v) => (k.name, fk.write(v)) }.asJava
```

Comment:
When I'm unpacking and repacking a tuple, I like the symmetry of having the bracket form on both sides.  If you could have the `->` sugar in the pattern match, then I'd do that.